src/cmd/run.rs

```diff
@@ -1,23 +1,37 @@
+use crate::adapters::BackendClient;
 use crate::adapters::backend::{ApplicationId, CreateRolloutParams, WorkspaceId};
 use crate::adapters::{BoxedIngress, BoxedMonitor, BoxedPlatform, RolloutMetadata};
+use crate::config::RunSubcommand;
 use crate::fs::{FileSystem, SessionFile, application_manifest};
 use crate::manifest::Manifest;
-use crate::subsystems::CONTROLLER_SUBSYSTEM_NAME;
-use crate::{ControllerSubsystem, adapters::BackendClient, config::RunSubcommand};
+use async_trait::async_trait;
 use miette::{Context, Diagnostic, Result, miette};
 use multitool_sdk::models::{ApplicationDetails, WorkspaceSummary};
 use thiserror::Error;
 use tokio::join;
 use tokio::runtime::Runtime;
-use tokio::time::Duration;
-use tokio_graceful_shutdown::{IntoSubsystem as _, SubsystemBuilder, Toplevel};
 use tracing::{debug, error, info};
 
 use crate::Terminal;
 
+use super::run_canary_mode::CanaryMode;
+use super::run_force_mode::ForceMode;
+
 /// The amount of time, in miliseconds, each subsystem has
 /// to gracefully shutdown before being forcably shutdown.
-const DEFAULT_SHUTDOWN_TIMEOUT: u64 = 5000;
+pub(super) const DEFAULT_SHUTDOWN_TIMEOUT: u64 = 5000;
+
+/// Trait defining different deployment modes for the MultiTool CLI
+#[async_trait]
+pub(super) trait DeploymentMode {
+    async fn handle(
+        backend: BackendClient,
+        monitor: BoxedMonitor,
+        ingress: BoxedIngress,
+        platform: BoxedPlatform,
+        meta: RolloutMetadata,
+    ) -> Result<()>;
+}
```

Comment:
I believe these should not be provided as part of the `handle` method, but each implementation should accept some/all/none of them when they are constructed. For example, both the `CanaryMode` and `ForceMode` types need these when they're constructed, but a `MockMode` type wouldn't.